src/Sarif.Converters/PyLintConverter.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis.Sarif.Writers;
+using Microsoft.CodeAnalysis.Sarif.Converters.PyLintObjectModel;
+
+namespace Microsoft.CodeAnalysis.Sarif.Converters
+{
+    public class PyLintConverter : ToolFileConverterBase
+    {
+        public override void Convert(Stream input, IResultLogWriter output, LoggingOptions loggingOptions)
+        {
+            input = input ?? throw new ArgumentNullException(nameof(input));
+            output = output ?? throw new ArgumentNullException(nameof(output));
+
+            PyLintLogList pyLintDefects = PyLintLogReader.ReadLog(input);
+
+            Tool tool = new Tool
+            {
+                Name = "PyLint",
+                FullName = "PyLint Static Code Analysis"
+            };
+
+            output.Initialize(id: null, automationId: null);
+            output.WriteTool(tool);
+
+            var results = new List<Result>();
+
+            foreach (PyLintLog entry in pyLintDefects)
+            {
+                results.Add(this.PrepareSarifLogEntry(entry));
+            }
+
+            var fileInfoFactory = new FileInfoFactory(MimeType.DetermineFromFileExtension, loggingOptions);
+            Dictionary<string, FileData> fileDictionary = fileInfoFactory.Create(results);
+
+            if (fileDictionary?.Any() == true)
+            {
+                output.WriteFiles(fileDictionary);
+            }
+
+            output.OpenResults();
+            output.WriteResults(results);
+            output.CloseResults();
+        }
+
+        internal Result PrepareSarifLogEntry(PyLintLog defect)
+        {
+            defect = defect ?? throw new ArgumentNullException(nameof(defect));
+
+            Result result = new Result()
+            {
+                RuleId = $"{defect.MessageId}({defect.Symbol})",
+                Message = defect.Message
+            };
+
+            switch (defect.Type)
+            {
+                case "error":
+                case "fatal":
+                    result.Level = ResultLevel.Error;
+                    break;
+                case "warning":
+                    result.Level = ResultLevel.Warning;
+                    break;
+
+                case "convention":
+                case "refactor":
+                    result.Level = ResultLevel.Default;
```

Comment:
IMO the mapping should be:

`"convention"` => `ResultLevel.Warning`
`"refactor"` => `ResultLevel.Note`
`default` => `ResultLevel.Note`

@michaelcfanning Do you agree?